src/Sulu/Component/Rest/ListBuilder/Metadata/Provider/MetadataProvider.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Metadata\Provider;
+
+use Metadata\MetadataFactoryInterface;
+use Sulu\Component\Rest\ListBuilder\Metadata\ProviderInterface;
+
+/**
+ * Provides metadata with the given metadata-factory.
+ */
+class MetadataProvider implements ProviderInterface
+{
+    /**
+     * @var MetadataFactoryInterface
+     */
+    private $metadataFactory;
+
+    public function __construct(MetadataFactoryInterface $metadataFactory)
+    {
+        $this->metadataFactory = $metadataFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMetadataForClass($className)
+    {
+        return $this->metadataFactory->getMetadataForClass($className);
```

Comment:
this is not the only class which implements this interface. see the ChainProvider which combines multiple other providers. so if we want to define this under one interface we have to delegate this. on the other hand it would also be possible to add other functions here like `getMetadataForObject` for example.